Customizing/global/plugins/Services/Repository/RepositoryObject/ReportExamBio/classes/class.ilObjReportExamBio.php

```diff
@@ -0,0 +1,323 @@
+	}
+
+	public function initSpace() {
+		$aux = $this->tf->histUsertestrun('recent_pass_aux');
+		$aux = $aux->addConstraint($aux->field('hist_historic')->EQ()->int(0)
+			->_AND(count($this->relevantUsers()) > 0 ? $aux->field('usr_id')->IN($this->pf->list_int_by_array($this->relevantUsers())) : $this->pf->_FALSE())
+			);
+
+		$recent_pass_case = $this->tf->derivedTable(
+			$this->tf->TableSpace()
+				->addTablePrimary($aux)
+				->setRootTable($aux)
+				->request($aux->field('usr_id'))
+				->request($aux->field('obj_id'))
+				->request($this->tf->maxSql('recent_pass',$aux->field('pass')))
+				->groupBy($aux->field('usr_id'))
+				->groupBy($aux->field('obj_id')),'recent_pass_case');
+
+		$all_pass = $this->tf->histUsertestrun('all_pass');
+		$all_pass = $all_pass->addConstraint($all_pass->field('hist_historic')->EQ()->int(0));
+
+		$recent_pass_data = $this->tf->histUsertestrun('recent_pass_data');
+		$recent_pass_data = $recent_pass_data->addConstraint($recent_pass_data->field('hist_historic')->EQ()->int(0));
+
+		$usr = $this->tf->histUser('usr');
+		$usr = $usr->addConstraint($usr->field('hist_historic')->EQ()->int(0)
+			->_AND(count($this->relevantUsers()) > 0 ?
+				$usr->field('user_id')->IN($this->pf->list_int_by_array($this->relevantUsers())) : $this->pf->_FALSE()));
+
+		$orgus = $this->tf->allOrgusOfUsers('orgu_all',$this->relevantUsers());
+
+		$this->space = $this->tf->TableSpace()
+			->addTablePrimary($recent_pass_case)
+			->addTableSecondary($recent_pass_data)
+			->addTableSecondary($all_pass)
+			->addTableSecondary($orgus)
+			->addTableSecondary($usr)
+			->setRootTable($recent_pass_case)
+			->addDependency($this->tf->TableJoin($recent_pass_case,$all_pass,
+				$recent_pass_case->field('usr_id')->EQ($all_pass->field('usr_id'))
+					->_AND($recent_pass_case->field('obj_id')->EQ($all_pass->field('obj_id'))))
+				)
+			->addDependency($this->tf->TableJoin($recent_pass_case,$usr,
+				$recent_pass_case->field('usr_id')->EQ($usr->field('user_id')))
+				)
+			->addDependency($this->tf->TableJoin($recent_pass_case,$orgus,
+				$recent_pass_case->field('usr_id')->EQ($orgus->field('usr_id')))
+				)
+			->addDependency($this->tf->TableJoin($recent_pass_case,$recent_pass_data,
+				$recent_pass_case->field('usr_id')->EQ($recent_pass_data->field('usr_id'))
+					->_AND($recent_pass_case->field('obj_id')->EQ($recent_pass_data->field('obj_id')))
+					->_AND($recent_pass_case->field('recent_pass')->EQ($recent_pass_data->field('pass'))))
+				)
+			->request($recent_pass_data->field('usr_id'))
+			->request($recent_pass_data->field('obj_id'))
+			->groupBy($recent_pass_case->field('usr_id'))
+			->groupBy($recent_pass_case->field('obj_id'));
+	}
+
+	public function applyFilterToSpace($filter_values) {
+		$settings = call_user_func_array(array($this->filter(), "content"), $filter_values)[0];
+
+		$predicate = current($settings);
+		while($sub_predicate = next($settings)) {
+			$predicate = $predicate->_AND($sub_predicate);
+		}
```

Comment:
This could be done in the filter already, i think. The decision, that the filters is _AND-combined to me seems to belong to the filter and not the application to the space.